src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -106,12 +106,22 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
     clr_path = probe_paths.coreclr;
     if (clr_path.empty() || !pal::realpath(&clr_path))
     {
-        trace::error(_X("Could not resolve CoreCLR path. For more details, enable tracing by setting COREHOST_TRACE environment variable to 1"));
-        return StatusCode::CoreClrResolveFailure;
-    }
+        // in a single-file case we may not need coreclr,
+        // otherwise fail early.
+        if (!bundle::info_t::is_single_file_bundle())
```

Comment:
On Windows we still require coreclr.dll even if single-file is on, right? So we should probably only allow this for non-Windows...

Reply:
Situation with Windows is such that it seems certain that shipping version of singlefile will require coreclr.dll, however we do not want to close the door to allowing "superhost" there. 
At very least for internal/experimental use. This is why I want to keep this the same between Unix/Windows. 

This check is an early failure check. We will still fail if we need coreclr.dll later and it is not there.